src/hotspot/share/runtime/synchronizer.cpp

```diff
@@ -1177,6 +1180,12 @@ void ObjectSynchronizer::inc_in_use_list_ceiling() {
   Atomic::add(&_in_use_list_ceiling, (jint)AvgMonitorsPerThreadEstimate);
 }
 
+void ObjectSynchronizer::set_in_use_list_ceiling(size_t new_value) {
+  // _in_use_list_ceiling is a jint so this cast could lose precision,
+  // but in reality the ceiling should never get that high.
+  _in_use_list_ceiling = (jint)new_value;
+}
```

Comment:
I'm sure I must have asked this before but why is AvgMonitorsPerThreadEstimate typed as `intx` rather than `int`? A `jint is 32-bit and so is `int` on all our 32-bit and 64-bit platforms; whereas `intx` is 64-bit on a 64-bit system.

And if we only ever set this field once why introduce a seemingly general use setter function instead of just doing the initialization directly in the initialize() method?

Reply:
@dholmes-ora - Thanks for the review.

The `intx` type was copied from some other option when `AvgMonitorsPerThreadEstimate`
was introduced. I'd have to look to see why `intx` is used instead of `int` in options.

set_in_use_list_ceiling() is used in this fix and also in the follow on fix: JDK-8226416.

Reply:
$ grep -c 'product(int,' src/hotspot/share/runtime/globals.hpp
1

$ grep -c 'product(intx,' src/hotspot/share/runtime/globals.hpp
68

Looks like we have 1 `int` option and 68 `intx` options. I'll look at changing
AvgMonitorsPerThreadEstimate from `intx` to `int`.